draft-ietf-quic-http.md

```diff
@@ -432,17 +409,76 @@ either by identifying the stream that carries a request or by using a Push ID
 Only a client can send PRIORITY frames.  A server MUST NOT send a PRIORITY
 frame.
 
+## Unidirectional Streams
+
+Unidirectional streams in both directions possess a stream header that indicates
+the stream's purpose.  Some purposes are defined in this document; other
+purposes MAY be defined by extensions to HTTP/QUIC.
+
+~~~~~~~~~~ drawing
+ 0 1 2 3 4 5 6 7
++-+-+-+-+-+-+-+-+
+|Stream Type (8)|
++-+-+-+-+-+-+-+-+
+~~~~~~~~~~
+{: #fig-stream-header title="Unidirectional Stream Header"}
+
+The stream header consists of a single byte, indicating the type of data to
+follow.  The format and structure of that data are defined by the specification
+of the stream type.
+
+If the header indicates a stream type which is not supported by the recipient,
+the stream MUST be aborted (using a QUIC STOP_SENDING frame) with an error of
+type HTTP_UNKNOWN_STREAM_TYPE.  The remainder of the stream SHOULD NOT be read
+by the recipient.
+
+###  Control Streams
 
-## Server Push
+The control stream is indicated by a stream type of `0x43`.  Data on this stream
+consists of HTTP frames, as defined in {{frames}}.
 
-HTTP/QUIC supports server push in a similar manner to {{!RFC7540}}, but uses
+Each side MUST initiate a single control stream at the beginning of the
+connection and send its SETTINGS frame as the first frame on this stream.  Only
+one control stream is permitted; receipt of a second stream which claims to be a
+control stream MUST be treated as a connection error of type
+HTTP_TOO_CONTROLLING.  If the control stream is closed at any point, this MUST
+be treated as a connection error of type HTTP_CLOSED_CRITICAL_STREAM.
```

Comment:
Have you considered creating a special stream for "settings", that is closed when all settings-related information is complete?  That would allow for sending MAX_PUSH_ID there with some certainty about status.  You could then reserve the "control" stream type for PRIORITY, MAX_PUSH_ID, CANCEL_PUSH, and GOAWAY frames.

Reply:
Reasonable suggestion; let's maybe do that as a second stage, though?

Reply:
Of course :)